### PR TITLE
Virtualbox 5.2 support

### PR DESCRIPTION
**Pull request by jehiah**
This adds support for VirtualBox 5.2 (now in [beta](https://forums.virtualbox.org/viewtopic.php?f=15&t=84286))

I've modeled this changes here after #7574 and subsequent refactoring of the 5.1 addition in #8326

### CODE REVIEW

**Comment by briancain** [comment]
I think this addition needs to be [autoloaded](https://github.com/hashicorp/vagrant/blob/master/plugins/providers/virtualbox/plugin.rb#L52-L60) to prevent the tests from failing

**Comment by jehiah** [comment]
ahh thanks for the tip @briancain. Added that and if tests pass i'll combine the two commits here.

**Comment by briancain** [comment]
@jehiah - hmm, that didn't seem to fix the tests, but I think the change was necessary anyway.

File: plugins/providers/virtualbox/driver/version_5_2.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require File.expand_path("../version_5_1", __FILE__)
+
+module VagrantPlugins
+  module ProviderVirtualBox
+    module Driver
+      # Driver for VirtualBox 5.2.x
+      class Version_5_1 < Version_5_1
```
Comment (briancain) [inline plugins/providers/virtualbox/driver/version_5_2.rb]: Ah @jehiah This should be:

```ruby
class Version_5_2 < Version_5_1
```

Comment (jehiah) [reply plugins/providers/virtualbox/driver/version_5_2.rb]: 🤦‍♂️

**Comment by jehiah** [comment]
@chrisroberts @briancain Thanks for the review here. Is there anything else i can do to help land this PR?

**Comment by maxkaplan** [comment]
Hey - Any idea when this update will be part for the download found here (https://www.vagrantup.com/downloads.html) - I'm still getting the does not support 5.2 error.

Thanks!

**Comment by briancain** [comment]
Hey @maxkaplan - our plan is to roll out a release of Vagrant by next week, which should include the updates in this pull request for supporting VirtualBox 5.2. Thanks!

**Comment by maxkaplan** [comment]
@briancain - Thanks for the update! 
I was a bit confused.

I'm sure you probability saw this [https://blogs.oracle.com/virtualization/vbox52](https://blogs.oracle.com/virtualization/vbox52) got posted to hackernews so you might have more people who are having issues with 5.2 compatibly. 

Thanks!

**Pull request merged**